### PR TITLE
DI-243 Dlq for undelivered change requests

### PR DESCRIPTION
**Pull request by mabe13**
## Link to JIRA Ticket

- https://nhsd-jira.digital.nhs.uk/browse/DI-243

## Description

Please include a summary of the change

### Noteworthy Changes

- These are changes the reviewer should look out for

## Type of change

Delete not appropriate

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## Testing

Please tick the testing that has been completed

- [ ] Unit tests
- [ ] Integration tests

## Developer Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have run the [code formatting checks](../README.md#code-quality)
- [ ] I have run the [code quality checks](../README.md#code-quality)
- [ ] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [ ] Unit test code coverage is at or above 80%
- [ ] New and existing unit tests pass locally with my changes
- [ ] Tests have added that prove my fix is effective or that my feature works (Integration tests)
- [ ] I have made corresponding changes to the documentation
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [ ] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

**Review by ed-sparkes — DISMISSED** [review]

**Review by ed-sparkes — DISMISSED** [review]

File: application/eventbridge_dlq_handler/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+aws-lambda-powertools
+requests
```
Comment (JackPlowman) [inline application/eventbridge_dlq_handler/requirements.txt:2]: Why is requests needed?

File: deployment/serverless.yml [hunk 2]
```diff
@@ -129,3 +129,17 @@ functions:
...
+    image: ${env:AWS_SAME_ACCOUNT_DOCKER_REGISTRY}/eventbridge-dlq-handler:${env:VERSION}
+    architecture: x86_64
+    name: ${env:PROJECT_ID}-${sls:stage}-eventbridge-dlq-handler
+    description: ${sls:stage} EventBridge DLQ Handler lambda
+    memorySize: 128
+    timeout: 30
+    role: arn:aws:iam::${env:AWS_ACCOUNT_ID}:role/${env:TF_VAR_eventbridge_dlq_handler_role_name}
+    tags:
+      PYTHON_VERSION: ${env:SERVERLESS_PYTHON_VERSION_TAG}
+    environment:
```
Comment (JackPlowman) [inline deployment/serverless.yml:143]: Don't need `CHANGE_EVENTS_TABLE_NAME` environment variable as not connected to dynamodb

File: infrastructure/stacks/lambda-iam-roles/main.tf [hunk 3]
```diff
@@ -278,3 +278,83 @@ resource "aws_kms_alias" "signing_key" {
...
+      "Effect": "Allow",
+      "Action": [
+        "sqs:DeleteMessage",
+        "sqs:GetQueueAttributes",
+        "sqs:ReceiveMessage"
+      ],
+      "Resource":"arn:aws:sqs:${var.aws_region}:${var.aws_account_id}:uec-dos-int-*"
+    },
+    {
+      "Effect": "Allow",
```
Comment (JackPlowman) [inline infrastructure/stacks/lambda-iam-roles/main.tf:338]: Don't need dynamodb policies as not connecting to dynamodb

**Review by JackPlowman — CHANGES_REQUESTED** [review]
Some small changes but overall looks good

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

**Review by ed-sparkes — APPROVED** [review]

**Pull request merged**